parser/logfmt/logfmt.go

```diff
@@ -115,21 +131,25 @@ func (p *Parser) parse(line string) ([]Data, error) {
 	reader := bytes.NewReader([]byte(line))
 	decoder := logfmt.NewDecoder(reader)
 	datas := make([]Data, 0)
+	var fields Data
 	for {
 		ok := decoder.ScanRecord()
 		if !ok {
 			err := decoder.Err()
 			if err != nil {
 				return nil, err
 			}
+			//此错误仅用于当原始数据解析成功但无解析数据时，保留原始数据之用
+			if len(fields) == 0 {
+				return nil, errors.New("field's length is zero,but original data should be reserved.")
```

Comment:
no value was parsed after logfmt, will keep origin data in pandora_stash if disable_record_errdata field is false